storybookjs/react-native
# feat: add annotations support for addons

Issue:

## What I did

Added support for preview files if they are present in the addon module

I required support for the storybook-addon-deep-controls addon, but it wasn't compatible with the React Native version of Storybook. This pull request will introduce support for this module and potentially others. Specifically for this addon, I've added a preview file to the root directory as it was previously only available in the dist directory. This modification will ensure compatibility and provide a foundation for integrating other modules in the future

<img width="338" alt="Screenshot 2024-10-01 at 17 29 37" src="https://github.com/user-attachments/assets/e3c296be-f37c-4933-9900-68ee92ae9b66">



<details>
  <summary>Deep Controls test</summary>

<img width="660" alt="Screenshot 2024-10-20 at 17 23 21" src="https://github.com/user-attachments/assets/67bf87ae-464c-4519-80e7-fd42d39a3108">

<img width="397" alt="Screenshot 2024-10-20 at 17 23 46" src="https://github.com/user-attachments/assets/4b47bb82-a099-4cfd-a242-018577446270">


</details>
## How to test

1. Add the addon which includes a preview enhancer
2. Check the generated content

<img width="442" alt="Screenshot 2024-10-01 at 17 04 02" src="https://github.com/user-attachments/assets/ea13baec-f028-4bc4-aec8-6bed54597e50">





- Does this need a new example in examples/expo-example? no
- Does this need an update to the documentation? no
